internal/controller/ws/client.go

```diff
@@ -204,33 +200,56 @@ func (c *rcClient) consume(ctx context.Context) {
 	}
 }
 
-func (c *rcClient) produce(ctx context.Context, m protocol.Message, kw *kafka.Writer) error {
+func (c *rcClient) produce(ctx context.Context, m protocol.Message) error {
 	type ResponseMessage struct {
-		Data struct {
-			Sender    string `json:"sender"`
-			Recipient string `json:"recipient"`
-			MessageID string `json:"in_response_to"`
-			Payload   string `json:"raw_payload"`
-		} `json:"Data"`
+		Account   string      `json:"account"`
+		Sender    string      `json:"sender"`
+		MessageID string      `json:"message_id"`
+		Payload   interface{} `json:"payload"`
 	}
 
-	r := ResponseMessage{}
+	if m.Type() != protocol.PayloadMessageType {
```

Comment:
At this point, I'm leaning toward having a single object that is responsible for "dispatching" the responses.  So whether the message is a ROUTE message or a PAYLOAD message, the reader will just tell the "Dispatcher" to just take the message and handle it however the dispatches see's fit.

I feel like it will be the "Dispatcher's" job to determine if the message is valid for routing or not.  The produce method is currently our "Dispatcher".

Reply:
Yep I thought about that after I commented :) Gonna merge